UnityProject/Assets/Scripts/Construction/Glass/IceShard.cs

```diff
@@ -19,18 +19,22 @@ private void Awake()
 
 	private void OnEnable()
 	{
-		UpdateManager.Add(UpdateCycle, 1f);
+		if (CustomNetworkManager.IsServer)
+		{
+			UpdateManager.Add(ServerUpdateCycle, 1f);
+		}
 	}
 
 	private void OnDisable()
 	{
-		UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, UpdateCycle);
+		if (CustomNetworkManager.IsServer)
+		{
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, ServerUpdateCycle);
+		}
 	}
 
-	private void UpdateCycle()
+	private void ServerUpdateCycle()
 	{
-		if(!CustomNetworkManager.IsServer) return;
-
 		var pos = registerTile.WorldPosition;
 
 		if (pos == posCache && metaDataNode.GasMix.Temperature > AtmosDefines.WATER_VAPOR_FREEZE)
```

Comment:
On the `else` after this can you change it to `else if (pos != posCache)` seems I forgot that optimisation.

Reply:
Done!